RFC/src/RFC-0310_AssetImplementation.md

```diff
@@ -0,0 +1,607 @@
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",
+"NOT RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as
+shown here.
+
+## Disclaimer
+
+This document and its content are intended for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Abstract
+
+## Related Requests for Comment
+
+None.
+
+## Problem Statement
+
+This RFC details the specifics of how NFTs are stored and represented on 
+the base layer and the digital assets layer.
+
+## Proposal
+
+### Changes to outputs
+The following fields must be included in the TransactionOutput, inside the OutputFeatures:
+
+New flags to be added to OutputFlags
+
+| Name | Value | Description |
+| --- | --- | --- |
+| NON_FUNGIBLE | 0b1000_1000 | This UTXO contains Non Fungile data and must not be combined with other fungible UTXOs |
+| ASSET_REGISTRATION | 0b1100_1000 | This UTXO contains registration information for a new asset |
+| MINT |0b0100_0000 | This UTXO represents the creation of a new NFT, and is used during the  |
+| BURN | 0b0010_0000 | The `unique_id` in this UTXO can be spent as a normal input and must not appear in the output set unless accompanied by a `MINT` flag |
+| SIDECHAIN_CHECKPOINT | 0b1001_1000 | This UTXO is a checkpoint for a sidechain | 
+
+
+New fields added to `OutputFeatures`
+
+Note: this replaces some information in [RFC-0311](RFC-0310_AssetImplementation.md)
+
+| Name | Type | Description |
+| --- | --- | --- |
+| unique_id | bytes(32) | A unique id representing a token. This value must be present only when the NON_FUNGIBLE flag is set. The pair (parent_pub_key, unique_id) will be unique across the unspent set as described below |
+| parent_pub_key | bytes(32) |  A namespacing field that also allows a heirachy of assets. In most cases this will be the asset that a token belongs to, but future uses may be possible as well |
```

Comment:
```suggestion
| parent_pub_key | bytes(32) |  A namespacing field that also allows a hierarchy of assets. In most cases this will be the asset that a token belongs to, but future uses may be possible as well |
```